examples/grpc/hello_server.cc

```diff
@@ -0,0 +1,137 @@
+#include "opencensus/exporters/stats/stackdriver/stackdriver_exporter.h"
+#include "opencensus/exporters/stats/stdout/stdout_exporter.h"
+#include "opencensus/exporters/trace/stackdriver/stackdriver_exporter.h"
+#include "opencensus/exporters/trace/stdout/stdout_exporter.h"
+#include "opencensus/plugins/grpc/grpc_plugin.h"
+#include "opencensus/trace/sampler.h"
+#include "opencensus/trace/span.h"
+#include "opencensus/trace/trace_config.h"
+#include "prometheus/exposer.h"
+
+namespace {
+
+using examples::HelloReply;
+using examples::HelloRequest;
+using examples::HelloService;
+
+// A helper function that performs some work in its own Span.
+void PerformWork(opencensus::trace::Span* parent) {
+  auto span = opencensus::trace::Span::StartSpan("internal_work", parent);
+  span.AddAttribute("my_attribute", "blue");
+  span.AddAnnotation("Performing work.");
+  absl::SleepFor(absl::Milliseconds(20));  // Working hard here.
+  span.End();
+}
+
+class HelloServiceImpl final : public HelloService::Service {
+  grpc::Status SayHello(grpc::ServerContext* context,
+                        const HelloRequest* request,
+                        HelloReply* reply) override {
+    opencensus::trace::Span span =
+        opencensus::GetSpanFromServerContext(context);
+    span.AddAttribute("my_attribute", "red");
+    span.AddAnnotation(
+        "Constructing greeting.",
+        {{"name", request->name()}, {"name length", request->name().size()}});
+    reply->set_message(absl::StrCat("Hello ", request->name(), "!"));
+    absl::SleepFor(absl::Milliseconds(10));
+    PerformWork(&span);
+    span.AddAnnotation("Sleeping.");
+    absl::SleepFor(absl::Milliseconds(30));
+    // TODO: Record() custom stats.
+    std::cerr << "SayHello RPC handled.\n";
+    return grpc::Status::OK;
+  }
+};
+
+}  // namespace
+
+int main(int argc, char** argv) {
+  // Handle port argument.
+  int port = 0;
+  if (argc == 2) {
+    if (!absl::SimpleAtoi(argv[1], &port)) {
+      std::cerr << "Invalid port number: \"" << argv[1] << "\"";
+      return 1;
+    }
+  }
+
+  // Register the OpenCensus gRPC plugin to enable stats and tracing in gRPC.
+  opencensus::RegisterGrpcPlugin();
+
+  // Register exporters for Stackdriver.
+  RegisterStackdriverExporters();
+
+  // For debugging, register exporters that just write to stdout.
+  opencensus::exporters::stats::StdoutExporter::Register();
+  opencensus::exporters::trace::StdoutExporter::Register();
+
+  // Keep a shared pointer to the Prometheus exporter.
+  auto exporter =
+      std::make_shared<opencensus::exporters::stats::PrometheusExporter>();
+
+  // Expose a Prometheus endpoint.
+  prometheus::Exposer exposer("127.0.0.1:8080");
+  exposer.RegisterCollectable(exporter);
+
+  // Add views for RPC stats.
+  const auto server_request_bytes_descriptor =
+      opencensus::stats::ViewDescriptor()
+          .set_measure(opencensus::kRpcServerRequestBytesMeasureName)
+          .set_name("server_request_bytes")
```

Comment:
One nit--we have been encouraging uri-style names specifying aggregation and tags, e.g. opencensus.io/server_request_bytes-distribution-method.

Reply:
Per discussion in meeting: example.com/...

Also add comment.

Reply:
This code was removed in favor of RegisterExperimentalEtc. I'll send a separate PR for custom views.